File: Searching/Binary Search/O(nlogn)_time_and_O(1)_space.js
function binarySearch(array, target) {
  // Write your code here.
  let start = 0;
  let end = array.length - 1;
  while (start <= end) {
    let middle = Math.floor((start + end) / 2);
    if (array[middle] === target) {
      return middle;
    } else if (array[middle] > target) {
      end = middle - 1;
    } else {
      start = middle + 1;
    }
  }
  return -1;
}

File: Searching/Binary Search/O(nlogn)_time_and_O(1)_space.py
def binarySearch(array, target):
    start=0
    end=len(array)-1
    while end>=start: 
        midInd=(start+end)//2
        if array[midInd] == target: 
            return midInd
        elif array[midInd] > target: 
            end=midInd-1
        else: 
            start=midInd+1
        print(start,end)
    return -1